ompi/mpi/fortran/use-mpi-f08/cdesc/pack_cdesc.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2005 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2011-2012 Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2015-2019 Research Organization for Information Science
+ *                         and Technology (RIST).  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "ompi/mpi/fortran/mpif-h/bindings.h"
+#include "ompi/mpi/fortran/use-mpi-f08/cdesc/bindings.h"
+#include "ompi/mpi/fortran/base/constants.h"
+
+void ompi_pack_cdesc(CFI_cdesc_t* x, MPI_Fint *incount, MPI_Fint *datatype,
+		     char *outbuf, MPI_Fint *outsize, MPI_Fint *position,
```

Comment:
Can't both the inbuf and outbuf of all these pack/unpack routines be descriptors?

I.e., don't we have to combine the inbuf, the MPI datatype, and the outbuf to do the underlying memcpy(ies) correctly?

Reply:
this is where I asked for guidance in a previous comment.

IMHO, that falls into the category "the enduser wants to do something pretty dumb but that is allowed by the standard".

Anyway, if we make `outbuf` a descriptor, then we cannot simply invoke `PMPI_Pack()` anymore (since it expects a contiguous `outbuf`), so it would be much easier to let the Fortran runtime handle this ... unless the compiler does not support `ignore_tkr`.

If simply returning an error when `outbuf` is not contiguous is not an option **and** we want to support compilers that cannot do `ignore_tkr`, then the only option is to do it by ourself.

Reply:
I do not think that there is a restriction on the outbuf to be contiguous -- i.e., it can be a descriptor.  More comments below in the main thread...